### PR TITLE
Rebar 3 support

### PR DESCRIPTION
**Pull request by arpunk**
This removes the old Makefiles and manual AMQP library installation in favor of rebar3 based compilation.

It also bumps the libraries to the latests published versions.

### CODE REVIEW

**Comment by oubiwann** [comment]
Wow, this looks great -- thanks!

One request, and then I'll merge: could you re-add the `lfe.config` file? (but you can remove the `depss` config directive).

Eventually, lfe projects won't have a src/*.app.src, but instead one will be generated for them from their `lfe.config` file. I think there's only one project that does that right now, but more will be able to do it easily once the lcfg rebar3 plugin is made ...

**Comment by arpunk** [comment]
@oubiwann Sure thing. I was under the impression lcfg was only used for skipping rebar dependencies and this is now handled properly with rebar3. Do you plan to just translate the metadata file to the `.app.src` file?

File: src/kanin.app.src [hunk 1]
```diff
@@ -1,41 +1,23 @@
...
-  %% is easier to include it in the .rel file followed by the atom 'none'
-  {included_applications, []},
-
-  %% configuration parameters similar to those in the config file specified
-  %% on the command line. can be fetched with gas:get_env
-  {env, []}
- ]
-}.
+  {maintainers, ["LFEx"]},
+  {licenses, [""]},
```
Comment (arpunk) [inline src/kanin.app.src:21]: @oubiwann I'm not sure what to fill here, can you help me out with this?

**Comment by oubiwann** [comment]
Yeah, the project that is currently generating it's `.app.src` file using lcfg does just that: writes the metadata out in Erlang syntax, having read it from its `lfe.config` file.

**Comment by oubiwann** [comment]
Thanks again!

**Comment by oubiwann** [comment]
Whoops -- I pulled the trigger too quickly!

File: src/kanin.app.src (hunk 1 above)
Comment (oubiwann) [inline src/kanin.app.src:21]: Do you mean maintainers and license? LFEx is fine for maintainers ... license should be the standard one that Erlang is moving to -- the Apache 2.0 License (previously discussed with Billo Systems).

Comment (arpunk) [inline src/kanin.app.src:21]: Ok, I will also add a `LICENSE` file for being extra compilant.

**Comment by arpunk** [comment]
It's ok, I will file up a new PR with the missing stuff by the end of the day.

**Pull request merged**